pkg/generators/forwarding/fluentd/templates.go

```diff
@@ -590,6 +591,50 @@ const pipelineToOutputCopyTemplate = `{{- define "pipelineToOutputCopyTemplate"
 </label>
 {{- end}}`
 
+const outputLabelConfCloudwatch = `{{- define "outputLabelConfCloudwatch" -}}
+<label {{.LabelName}}>
+  <filter kubernetes.**>
+    @type record_transformer
+	enable_ruby true
+    <record>
+      cw_group_name {{.LogGroupPrefix }}{{.LogGroupName }}
+      cw_stream_name ${tag}
+    </record>
+  </filter>
+  <filter journal **_default_** **_kube-*_** **_openshift-*_** **_openshift_**>
+    @type record_transformer
+	enable_ruby true
+	<record>
+      cw_group_name {{.LogGroupPrefix }}infrastructure
+      cw_stream_name ${record['hostname']}.${tag}
+    </record>
+  </filter>
+  <filter *audit.log>
+    @type record_transformer
+	enable_ruby true
+    <record>
+      cw_group_name {{.LogGroupPrefix }}audit
+      cw_stream_name ${record['hostname']}.${tag}
+    </record>
+  </filter>
+  <match **>
+    @type cloudwatch_logs
+    auto_create_stream true
+    region {{ .Target.Cloudwatch.Region }}
+    log_group_name_key cw_group_name
+	log_stream_name_key cw_stream_name
+    remove_log_stream_name_key true
+    remove_log_group_name_key true
+    auto_create_stream true
+    concurrency 2
+    aws_key_id "#{open('{{ .SecretPath "aws_access_key_id"}}','r') do |f|f.read end}"
```

Comment:
Only set this if the key is found, otherwise we'll have a new  crash loop when its missing. Here's what I did for the general TLS stuff:

```
{{- with $path := .SecretPathIfFound "tls.key"}}
tls_client_private_key_path "{{$path}}"
{{- end}}
```

Also if these keys are required, they should be checked as part of VerifyOutputSecret, but only for cloudwatch outputs.

Reply:
These are absolutely required for the plugin to work.  Will the generator code prior to the template successfully skip output generation for missing secrets?

Reply:
added